### PR TITLE
Update web font

### LINKED ISSUE #933: Better web font for I (capital i) and l (small L)
**Description of the problem**

See this page as an example: https://pygmt-6lvxfpcmx.vercel.app/gallery/line/vector-heads-tails.html

image

The two "I" in the two red boxes are different. The first one is capital `i`, and the second one is small `L`. However, they look the same in the current font (I'm using Microsoft Edge on macOS. Not sure what they look like on other web browsers and platforms).

Should we switch to a better web font? Does the current theme "sphinx_rtd_theme" support custom web fonts?

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR serves to preview different web font options based on the discussion at #933.

Fixes #933 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Looks good with Microsft Edge, Google Chrome, Safari, and Firefox on macOS.

**Comment by liamtoney** [comment]
From my limited knowledge of web design, I think we could serve the actual webfont file and then it should work regardless of platform.

**Comment by liamtoney** [comment]
[Google Fonts](https://fonts.google.com/?category=Sans+Serif) makes it really easy to test sentences.

File: doc/_static/style.css [hunk 1]
```diff
@@ -3,12 +3,12 @@ html {
 }
 
 body {
-    font-family: 'Open Sans', sans-serif;
+    font-family: 'Tahoma', 'Open Sans', sans-serif;
```
Comment (liamtoney) [inline doc/_static/style.css:6]: I think we want to fall back to `sans-serif`, not our old font choice.
```suggestion
    font-family: 'Tahoma', sans-serif;
```

File: doc/_static/style.css [hunk 2]
```diff
@@ -3,12 +3,12 @@ html {
...
 }
 
 h1, h2, h3, h4 {
     font-weight: 300;
-    font-family: "Open Sans",sans-serif;
+    font-family: "Tahoma", "Open Sans",sans-serif;
```
Comment (liamtoney) [inline doc/_static/style.css:11]:
```suggestion
    font-family: "Tahoma", sans-serif;
```

**Comment by liamtoney** [comment]
What about IBM Plex Sans? It differentiates lower-case L and upper-case I even more:

Screen Shot 2021-02-19 at 11 41 07 AM

To use one of the many Google Fonts, we could either:
1. Link in the HTML `<head>` (the below code is copied straight from the Google Fonts page):
    ```
    <link rel="preconnect" href="https://fonts.gstatic.com">
    <link href="https://fonts.googleapis.com/css2?family=IBM+Plex+Sans:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet">
    ```

2. Download the font and serve it.

In either case we don't have to bank on folks having the font installed locally, I think.

**Comment by meghanrjones** [comment]
I am not knowledgeable enough about licensing to know whether it's an option to ship fonts with PyGMT. In the python packages that I use most often, I have noticed fonts included for logos (e.g., matplotlib) but generally not for the main docs.

**Comment by liamtoney** [comment]
> I am not knowledgeable enough about licensing to know whether it's an option to ship fonts with PyGMT. In the python packages that I use most often, I have noticed fonts included for logos (e.g., matplotlib) but generally not for the main docs.

Yeah, I think option 1) is pretty standard and likely makes more since to keep the repo smaller etc. All Google Fonts are open-source: https://developers.google.com/fonts/faq#can_i_use_fonts_from_the_google_fonts_catalog_on_any_page

**Comment by seisman** [comment]
> 1. Link in the HTML `<head>` (the below code is copied straight from the Google Fonts page):
>    ```
>    <link rel="preconnect" href="https://fonts.gstatic.com">
>    <link href="https://fonts.googleapis.com/css2?family=IBM+Plex+Sans:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet">
>    ```

Most Google services and sites are blocked in China 😞. I'd appreciate it if we can avoid using these external font resources.

**Comment by liamtoney** [comment]
> > 1. Link in the HTML `<head>` (the below code is copied straight from the Google Fonts page):
> >    ```
> >    <link rel="preconnect" href="https://fonts.gstatic.com">
> >    <link href="https://fonts.googleapis.com/css2?family=IBM+Plex+Sans:ital,wght@0,400;0,700;1,400;1,700&display=swap" rel="stylesheet">
> >    ```
> 
> Most Google services and sites are blocked in China 😞. I'd appreciate it if we can avoid using these external font resources.

Shoot, I didn't realize. Good point.

**Comment by liamtoney** [comment]
I made a branch [`docs-fonts-serve`](https://github.com/GenericMappingTools/pygmt/tree/docs-fonts-serve) which demos what serving a font would be like. @seisman I guess we can't preview unless I turn it into a PR?

This setup does not use Google. IBM Plex Sans is licensed under [OFL](https://en.wikipedia.org/wiki/SIL_Open_Font_License), meaning it's free to use and serve etc. Many fonts fall into this category. Font files are about 300 KB in total so not much of an impact on repo size I think. Serving the font guarantees that all users see the same thing (I think).

Screen Shot 2021-02-19 at 1 04 58 PM

I think this looks pretty nice, but I'm a bit of an aesthete...

**Comment by seisman** [comment]
> I guess we can't preview unless I turn it into a PR?

Yes.

**Comment by liamtoney** [comment]
I opened #936 so we can get a preview. We can keep discussion here though (as I note in that PR).

**Comment by weiji14** [comment]
> I opened #936 so we can get a preview. We can keep discussion here though (as I note in that PR).
> 
> **Edit:**
> 
> Preview link: https://pygmt-ocupv4u4w.vercel.app/

I feel like the small letter I (i) looks too much like a small letter L (l), not enough spacing between the dot of the small letter i with the lower vertical part :sweat_smile:

image

**Comment by liamtoney** [comment]
I see what you mean. Maybe worse on lower-resolution screens? Though that could be a case for a lot of fonts.

**Pull request closed without merging**